src/jbuf/jbuf.c

```diff
@@ -408,16 +415,19 @@ int jbuf_put(struct jbuf *jb, const struct rtp_header *hdr, void *mem)
 	f->hdr = *hdr;
 	f->mem = mem_ref(mem);
 
-	/* Count not equal timestamp frames (e.g. video) */
-	if (f->le.prev) {
-		struct packet *pre_f = f->le.prev->data;
+	jb->nf = 1;
+	LIST_FOREACH(&jb->packetl, le)
```

Comment:
Is it possible to avoid `O(n)` effort for `jbuf_put()`?

Reply:
I have not found a reliable way. Out-of-order, delayed packets lead to double or missed frame counts, and incorrect frame counts have a big impact on latency and jitter calculations.

Reply:
The packets are inserted sorted by the sequence number. The sorted insert needs normally only one to a few checks.
Timestamps and sequence number should have the same order. So after the insert only one or a few checks are needed to find packets with same timestamp. Did I miss something?

Reply:
Draft: https://github.com/baresip/re/pull/821

Reply:
The problem is, we don't know if a sequence number is missing, because its lost or because its arrives later. This makes a simple increment unreliable. Maybe a list of missing sequence numbers can work. I plan to add `RTCP Feedback NACK` to trigger a resend for missing packets (video only).